### PR TITLE
Feature/3367 replace legacy tags

### PR DESCRIPTION
**Pull request by amoedoamorim**
## Description

Refactoring MediaTags with Relay Modern and making use of TagList cds component.

Changelist:
- Added onClick prop to Chip to allow click to search by tag
- Removed older MediaTags_projecMedia fragment references
- Refactored MediaTags with Relay Modern QueryRenderer and mutations
- Deprecated TagMenu component
- Deprecated relay classic tag mutations

References: CV2-3367

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Tested manually that adding and removing tags to an item works as expected.
Also clicking a tag to search by that tag also still works.

## Things to pay attention to during code review

- Are fragments in tags still relevant? Fragments were related to video tags, which it was used to tag segments of a video.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by amoedoamorim** [comment]
Pending fixing unit tests

**Comment by caiosba** [comment]
Answering questions in the PR: We don't need fragments anymore, it was only used by the video player.

File: src/app/components/media/MediaTags.js [hunk 1]
```diff
@@ -91,52 +62,122 @@ class MediaTags extends React.Component {
...
+    !can(projectMedia.permissions, 'update ProjectMedia') ||
+    projectMedia.is_secondary ||
+    projectMedia.suggested_main_item?.dbid ||
+    projectMedia.archived > CheckArchivedFlags.NONE;
+
+  const { regularTags, videoTags } = filterTags(projectMedia.tags.edges);
+  const tags = regularTags.concat(videoTags);
+
+  const onFailure = () => {
+    console.log('failed to create or delete tags'); // eslint-disable-line
```
Comment (caiosba) [inline src/app/components/media/MediaTags.js:86]: Can we replace this by the regular flash component?

File: src/app/components/media/MediaTags.js [hunk 2]
```diff
@@ -91,52 +62,122 @@ class MediaTags extends React.Component {
...
+    tagTexts.forEach((text) => {
+      if (!value.includes(text)) handleRemoveTag(text);
+    });
+    value.forEach((val) => {
+      if (!tagTexts.includes(val)) handleCreateTag(val);
+    });
+  };
+
+  // const selected = projectMedia.tags.edges.map(t => t.node.tag_text);
+  // const options = projectMedia.team.tag_texts.edges.map(tt => ({ label: tt.node.text, value: tt.node.text }));
```
Comment (caiosba) [inline src/app/components/media/MediaTags.js:167]: Do we need to keep these comments?

File: src/app/components/media/MediaTags.js [hunk 3]
```diff
@@ -156,26 +197,55 @@ MediaTags.propTypes = {
...
+          permissions
+          team {
+            slug
+            #  tag_texts(last: 100) {
+            #    edges {
+            #      node {
+            #        text
+            #      }
+            #    }
+            #  }
```
Comment (caiosba) [inline src/app/components/media/MediaTags.js:220]: Do we need to keep these comments?

**Review by caiosba — CHANGES_REQUESTED** [review]
I left a few comments.

**Comment by amoedoamorim** [comment]
@caiosba Ready for review

**Review by caiosba — APPROVED** [review]
Thanks for addressing the review comments!

**Pull request merged**